protected/humhub/modules/content/Events.php

```diff
@@ -143,6 +144,27 @@ public static function onCronBeforeAction($event): void
         }
     }
 
+    public static function onViewEndBody($event)
+    {
+        $highlight = Yii::$app->session->get('contentHighlight');
+        if ($highlight === null || $highlight === '') {
+            return;
+        }
+        Yii::$app->session->remove('contentHighlight');
+
+        /* @var View $view */
+        $view = $event->sender;
+
+        $index = time();
+        $view->registerJs('function contentHighlight' . $index . '() {
+            "' . str_replace('"', '\"', $highlight) . '".split(" ")
```

Comment:
I think we need better encoding here. I'm a bit afraid that this could lead to an injection into the JS that is not intercepted by the new nonce.


Can you please use something like 

`$this->registerJsVar('contentSearchKeyword', $keyword);` to safely pass the PHP Var to JS.

Reply:
@luke- Updated in the commit https://github.com/humhub/humhub/pull/6971/commits/55381159b7a3ad050ebd6d02aa56a7bff41c7733.